server/config/parser.js

```diff
@@ -58,6 +58,15 @@ module.exports = function( configPath, defaultOpts ) {
 		} );
 	}
 
+	if (
+		! ( secretsPath === realSecretsPath ) &&
+		data.features &&
+		data.features[ 'wpcom-user-bootstrap' ]
+	) {
+		console.error( 'Disabling server-side user-bootstrapping because of a missing secrets.json' );
+		data.features[ 'wpcom-user-bootstrap' ] = false;
+	}
+
```

Comment:
Hmm, the config parser is a place where I wouldn't expect this.
Would a viable alternative be to add these checks explicitly where we're now checking for `config.isEnabled( 'wpcom-user-bootstrap' )`?

Reply:
i agree it isn't ideal. the client doesn't have access to the secrets though :/

Reply:
Hmm, I've noticed `client/` side checks for `config.isEnabled( 'wpcom-user-bootstrap' )` now, but are those even necessary? We should be able to simply check if the server is giving us a `window.currentUser` object in bootstrapping -- possibly having the server set it to `false` if bootstrapping is off...

Reply:
(Fine to do in another PR I guess so we don't block the upgrade any longer.)

Reply:
> We should be able to simply check if the server is giving us a window.currentUser

thats a good idea! and would clean this up.

>  (Fine to do in another PR I guess so we don't block the upgrade any longer.)

that would be my vote.  I'll make an issue for it now.  I'm wary of diving down more rabbit holes right now in case that turns out to be more complicated than we expect.  I also wouldn't want to add even more fixes onto this pr.

Reply:
issue: https://github.com/Automattic/wp-calypso/issues/20308